lib/configurators/aws/aws_test.go

```diff
@@ -1935,10 +1943,69 @@ func TestExtractTargetConfig(t *testing.T) {
 				},
 			},
 		},
+		"manual mode role name target assuming config roles is an error": {
+			target: manualModeRoleTarget,
+			flags:  configurators.BootstrapFlags{ForceAssumesRoles: role1, Manual: true},
+			cfg: &servicecfg.Config{
+				Discovery: servicecfg.DiscoveryConfig{
+					AWSMatchers: []types.AWSMatcher{
+						{Types: []string{services.AWSMatcherRDSProxy}, AssumeRole: &types.AssumeRole{RoleARN: roleTarget.String()}},
+					},
+				},
+				Databases: servicecfg.DatabasesConfig{
+					Databases: []servicecfg.Database{
+						{Name: "db1", AWS: servicecfg.DatabaseAWS{AssumeRoleARN: role2}},
+					},
+					AWSMatchers: []types.AWSMatcher{
+						{Types: []string{services.AWSMatcherRDS, services.AWSMatcherEC2}, AssumeRole: &types.AssumeRole{RoleARN: role4}},
+					},
+				},
+			},
+			wantErr: "please specify the full role ARN",
+		},
+		"manual mode role name target assuming only forced roles is ok": {
+			target: manualModeRoleTarget,
+			flags:  configurators.BootstrapFlags{ForceAssumesRoles: role1, Manual: true},
+			cfg: &servicecfg.Config{
+				Databases: servicecfg.DatabasesConfig{
+					AWSMatchers: []types.AWSMatcher{
+						{Types: []string{services.AWSMatcherRDS, services.AWSMatcherEC2}},
+					},
+				},
+			},
+			want: targetConfig{
+				identity:        roleTarget,
+				assumesAWSRoles: []string{role1},
+				awsMatchers: []types.AWSMatcher{
+					{Types: []string{services.AWSMatcherRDS, services.AWSMatcherEC2}},
+				},
+			},
+		},
+		"manual mode role name target without any assume roles is ok": {
```

Comment:
I just realized the manual flag has no effect on `getTargetConfig`, since by then we've already determined the target, so really the full ARN case is covered by all the other test cases in `TestExtractTargetConfig`

Reply:
I added a case anyway [302b249](https://github.com/gravitational/teleport/pull/29381/commits/302b249c53b63f08f0686dc7b85c1243bad6ce61)